.github/workflows/test-ballerine-wf.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish workflows-service image
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      # Run this pipeline only if there are changes in specified path
+      - "services/workflows-service/**"
+    branches:
+      - dev
+      - test
+      - prod
+      - staging
+      - sb
+      - demo
+
+jobs:
+  ci-test:
+    runs-on: ubuntu-latest
+    defaults:
+        run:
+          shell: bash
+    # Service containers to run with this job. Required for running tests
+    services:
+        # Label used to access the service container
+        postgres:
+          # Docker Hub image for Redis
+          image: sibedge/postgres-plv8:15.3-3.1.7
+          ports:
+            # Opens tcp port 5432 on the host and service container
+            - 5432:5432
```

Comment:
have you check the option to use dind?(Docker in docker?)

So, it should be easy to just run docker compose file and than the command for test job 
https://github.com/peter-evans/docker-compose-actions-workflow